# Homework Assignment - Lesson 1: Introduction to R

**Student Name:** [Marcelo Coronel]  
**Date:** [9/8/2025]  
**Course:** Data Management  

---

## Instructions

Complete all the tasks below by adding your R code in the code cells and your written responses in markdown cells. Make sure to run all cells to show your output.

**📋 SUBMISSION**: When you're done, see [GITHUB_CLASSROOM_SUBMISSION.md](../../GITHUB_CLASSROOM_SUBMISSION.md) for complete submission instructions.

---

## Part 1: Setting Up Your Environment and Importing Data

### 1.1 Working Directory

Check your current working directory using `getwd()`. You should be in the main repository folder.

In [1]:
# Check current working directory
setwd('/workspaces/assignment-1-coronelmm')
getwd()

[1] "/workspaces/assignment-1-coronelmm"

### 1.2 Package Loading

Load the required packages: `tidyverse` and `readxl`

In [2]:
# Load required packages
library(tidyverse)
library(readxl)

── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.2     ✔ tibble    3.3.0
✔ lubridate 1.9.4     ✔ tidyr     1.3.1
✔ purrr     1.1.0     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors


### 1.3 Data Import - CSV

Import the `sales_data.csv` file from the `data/` folder into a data frame named `sales_df`.

In [3]:
# Import sales data
sales_df <- read.csv("data/sales_data.csv")
print("Sales data imported successfully!")

[1] "Sales data imported successfully!"


### 1.4 Data Import - Excel

Import both sheets from the `customer_feedback.xlsx` file.

In [4]:
# Import ratings data from Excel
ratings_df <- read_excel("data/customer_feedback.xlsx", sheet = "ratings")

# Import comments data from Excel  
comments_df <- read_excel("data/customer_feedback.xlsx", sheet = "customer_feedback")

print("Excel data imported successfully!")

[1] "Excel data imported successfully!"


---

## Part 2: Basic Data Inspection

For each of the three data frames you imported, perform the following inspection tasks. Add both your R code and a brief interpretation of what you observe.

### 2.1 Sales Data Inspection

In [5]:
# First 10 rows of sales data
print("First 10 rows of sales_df:")
head(sales_df, 10)

[1] "First 10 rows of sales_df:"


TransactionID Date       Product  Amount  Quantity Region
1   1            2023-01-27 Keyboard 1039.05 1        South 
2   2            2023-05-18 Tablet    597.55 5        East  
3   3            2023-02-14 Keyboard 1539.48 1        West  
4   4            2023-07-28 Mouse     261.66 1        South 
5   5            2023-05-17 Monitor  1047.73 1        South 
6   6            2023-02-25 Keyboard 1824.05 2        East  
7   7            2023-01-20 Monitor   475.83 1        North 
8   8            2023-07-14 Mouse     758.05 5        North 
9   9            2023-10-02 Mouse    1717.20 4        South 
10 10            2023-10-15 Laptop   1437.21 3        North

In [6]:
# Structure of sales data
print("Structure of sales_df:")
str(sales_df)

[1] "Structure of sales_df:"
'data.frame':	100 obs. of  6 variables:
 $ TransactionID: int  1 2 3 4 5 6 7 8 9 10 ...
 $ Date         : chr  "2023-01-27" "2023-05-18" "2023-02-14" "2023-07-28" ...
 $ Product      : chr  "Keyboard" "Tablet" "Keyboard" "Mouse" ...
 $ Amount       : num  1039 598 1539 262 1048 ...
 $ Quantity     : int  1 5 1 1 1 2 1 5 4 3 ...
 $ Region       : chr  "South" "East" "West" "South" ...


In [7]:
# Summary statistics for sales data
print("Summary statistics for sales_df:")
summary(sales_df)

[1] "Summary statistics for sales_df:"


 TransactionID        Date             Product              Amount       
 Min.   :  1.00   Length:100         Length:100         Min.   :  58.06  
 1st Qu.: 25.75   Class :character   Class :character   1st Qu.: 520.79  
 Median : 50.50   Mode  :character   Mode  :character   Median : 960.75  
 Mean   : 50.50                                         Mean   : 962.71  
 3rd Qu.: 75.25                                         3rd Qu.:1384.63  
 Max.   :100.00                                         Max.   :1985.60  
    Quantity       Region         
 Min.   :1.00   Length:100        
 1st Qu.:1.00   Class :character  
 Median :3.00   Mode  :character  
 Mean   :2.69                     
 3rd Qu.:4.00                     
 Max.   :5.00                     

**Your observations about sales_df:**

These seem to be sales data for different electronic products. The first one was head so it showed me the first 10 rows of the sales dataset. Each row represents a transaction with a TransactionID, the date, the product sold, the sales amount, the quantity purchased, and the sales region. For example, Transaction 1 was on 2023-01-27, selling a Keyboard in the South region for 1039.05 with a quantity of 1. The next one was str and it provided the structure of the dataset. It told me that sales_df is a data frame with 100 observations and 6 variables. TransactionID and Quantity are integers, Date, Product, and Region are character strings, and Amount is numeric. It also showed sample values for each column to preview the data. The last one was summary and it provided descriptive statistics. I think the most useful part was the Amount and Quantity columns. Amount ranged from about 58 to 1985 with a mean of 962.71. Quantity ranged from 1 to 5 with an average of 2.69. For the character variables. 

### 2.2 Ratings Data Inspection

In [8]:
# TODO: ratings_df inspection
print("First 10 rows of ratings_df:")
head(ratings_df, 10)
print("Structure of ratings_df:")
str(ratings_df)
print("Summary statistics for ratings_df:")
summary(ratings_df)

[1] "First 10 rows of ratings_df:"


CustomerID ProductRating ServiceRating OverallSatisfaction
1   1         1             4             3                  
2   2         3             2             4                  
3   3         1             5             5                  
4   4         1             2             3                  
5   5         5             1             2                  
6   6         4             2             2                  
7   7         3             5             5                  
8   8         5             2             5                  
9   9         4             1             3                  
10 10         4             1             2                  
   RecommendationScore
1   2                 
2   7                 
3   1                 
4   6                 
5   2                 
6   6                 
7   5                 
8  10                 
9   6                 
10  0

[1] "Structure of ratings_df:"
tibble [50 × 5] (S3: tbl_df/tbl/data.frame)
 $ CustomerID         : num [1:50] 1 2 3 4 5 6 7 8 9 10 ...
 $ ProductRating      : num [1:50] 1 3 1 1 5 4 3 5 4 4 ...
 $ ServiceRating      : num [1:50] 4 2 5 2 1 2 5 2 1 1 ...
 $ OverallSatisfaction: num [1:50] 3 4 5 3 2 2 5 5 3 2 ...
 $ RecommendationScore: num [1:50] 2 7 1 6 2 6 5 10 6 0 ...
[1] "Summary statistics for ratings_df:"


   CustomerID    ProductRating ServiceRating  OverallSatisfaction
 Min.   : 1.00   Min.   :1.0   Min.   :1.00   Min.   :1.00       
 1st Qu.:13.25   1st Qu.:2.0   1st Qu.:1.00   1st Qu.:2.00       
 Median :25.50   Median :3.0   Median :3.00   Median :3.00       
 Mean   :25.50   Mean   :3.2   Mean   :2.94   Mean   :2.92       
 3rd Qu.:37.75   3rd Qu.:5.0   3rd Qu.:5.00   3rd Qu.:4.00       
 Max.   :50.00   Max.   :5.0   Max.   :5.00   Max.   :5.00       
 RecommendationScore
 Min.   : 0.0       
 1st Qu.: 1.0       
 Median : 4.5       
 Mean   : 4.2       
 3rd Qu.: 7.0       
 Max.   :10.0       

**Your observations about ratings_df:**

These seem to be customer ratings for different products. The first one was head I could see that each row represents a customer (CustomerID 1–10) and the columns include their ProductRating, ServiceRating, OverallSatisfaction, and RecommendationScore. The values are numeric, for example Customer 1 rated Product = 1, Service = 4, Satisfaction = 3, and Recommendation = 2. The second one was str and it showed that all columns are numeric including customer_id, product_rating, service_rating, overall_satisfaction, and recommendation_score. The third one was summary so it provided a statistical summary of each variable in the data frame, for this one we see that recommendation_score is the highest central tendency around 4.3ish and the other ones had a lower central tendency around 3. Perhaps recommendation_score is higher because for that one they rated it 1 to 10 instead of 1 to 5 like the other ones.



### 2.3 Comments Data Inspection

In [9]:
# TODO: Add your code here for comments_df inspection
print("First 10 rows of comments_df:")
head(comments_df, 10)
print("Structure of comments_df:")
str(comments_df)
print("Summary statistics for comments_df:")
summary(comments_df)

[1] "First 10 rows of comments_df:"


FeedbackID CustomerID Feedback_Text                   
1   1         12         Highly recommend this item      
2   2         40         Excellent service               
3   3         34         Poor quality control            
4   4          1         average product, nothing special
5   5         47         AMAZING customer support!!!     
6   6         13         AMAZING customer support!!!     
7   7         13         average product, nothing special
8   8         37         good VALUE for money            
9   9         49         Highly recommend this item      
10 10         23         Highly recommend this item      
   Contact_Info           Feedback_Date
1  bob.wilson@test.org    2024-02-23   
2  555-123-4567           2024-01-21   
3  jane_smith@company.com 2023-09-02   
4  jane_smith@company.com 2023-08-21   
5  555-123-4567           2023-04-24   
6  john.doe@email.com     2023-04-16   
7  (555) 987-6543         2024-03-13   
8  (555) 987-6543         2023-07-15   
9  invalid-email          2023-03-03   
10 john.doe@email.com     2023-10-18

[1] "Structure of comments_df:"
tibble [100 × 5] (S3: tbl_df/tbl/data.frame)
 $ FeedbackID   : num [1:100] 1 2 3 4 5 6 7 8 9 10 ...
 $ CustomerID   : num [1:100] 12 40 34 1 47 13 13 37 49 23 ...
 $ Feedback_Text: chr [1:100] "Highly recommend this item" "Excellent service" "Poor quality control" "average product, nothing special" ...
 $ Contact_Info : chr [1:100] "bob.wilson@test.org" "555-123-4567" "jane_smith@company.com" "jane_smith@company.com" ...
 $ Feedback_Date: POSIXct[1:100], format: "2024-02-23" "2024-01-21" ...
[1] "Summary statistics for comments_df:"


   FeedbackID       CustomerID    Feedback_Text      Contact_Info      
 Min.   :  1.00   Min.   : 1.00   Length:100         Length:100        
 1st Qu.: 25.75   1st Qu.:15.00   Class :character   Class :character  
 Median : 50.50   Median :26.00   Mode  :character   Mode  :character  
 Mean   : 50.50   Mean   :26.27                                        
 3rd Qu.: 75.25   3rd Qu.:38.00                                        
 Max.   :100.00   Max.   :50.00                                        
 Feedback_Date                
 Min.   :2023-01-03 00:00:00  
 1st Qu.:2023-04-23 18:00:00  
 Median :2023-08-24 12:00:00  
 Mean   :2023-08-22 07:26:24  
 3rd Qu.:2023-12-07 06:00:00  
 Max.   :2024-03-30 00:00:00  

**Your observations about comments_df:**

This seems to be customer comments about their experience with the products or services. The first one was head so it showed the first 10 rows of the data frame, each row has a feedback_id, the customer_id it belongs to, the comment itself, and the date it was made some issues are spotted here like duplicate ids and an invalid email. The second one was str and it showed that feedback_id and customer_id are integers, comment is a character, and date is also a character. The third one was summary so it provided a statistical summary of each variable though these weren't as informative since feedback_id and customer_id are integers but they are identifiers, comment is a character so it provided length class and mode, and date is also a character so it provided length class and mode as well.

**Reflection Question 1:** What did you observe about the structure and content of the data files you loaded? What insights can you share about the quality and organization of the data?

**Your Answer:**

They seemed like real world applications of data. Each file had different types of information businesses might be interested in collecting. The sales data showed transaction details such as IDs, dates, products, amounts, quantities, and regions. The ratings data showed customer evaluations across product quality, service, satisfaction, and recommendation. The comments data provided customer feedback in text form along with contact information and the date of submission. To me, all three of these can be useful for a business to analyze and improve their operations.

### Question 1: Data Types Analysis

Based on your inspection of `sales_df`, what are the data types of the `Date` and `Amount` columns? Are these data types appropriate for typical business analytics tasks involving sales data? Explain why or why not.

So date and amount were different respectively. Date was character and amount was numeric. I think numeric values are more easily analyzed since you run calculations using numbers (with numeric values). Date being character is not as useful since you can't do calculations with characters. In this particular case, the date data was not appropriate for typical business analytics tasks. However, if there were a way to have date represent actual dates (instead of characters) then it would be more useful. For example, it might be useful to know that sales increased in December compared to November. But if date is just a character string, then you can't do that type of analysis.

### Question 2: Data Quality Assessment

Looking at all three datasets, what potential data quality issues do you notice? Consider missing values, data types, and any unusual patterns.

In the sales data, I think the date being a character is a potential data quality issue since it limits the types of analysis you can do with it. In the ratings data, all the variables are numeric, but the RecommendationScore uses a 0–10 scale while the other ratings use 1–5, which could lead to inconsistencies if compared directly (which I initially did). In the comments data, I noticed duplicate feedback IDs (13 appears twice), invalid email format, Same person rating multiple times and more. These seem like pretty significant data quality issues that would need to be addressed before analysis.

### Question 3: Analysis Readiness

Which of the three datasets appears most ready for analysis, and which would require the most preprocessing? Justify your answer.

**Your Answer:**

Ratings data seems the most ready for analysis since all the variables are numeric and there don't appear to be any glaring issues, and the one problem I mentioned earlier about the different rating scales can be easily addressed by using statistical techniques. Comments data would require the most preprocessing due to the presence of duplicate IDs, invalid email formats, and potential inconsistencies in the comments themselves. Cleaning and standardizing this text data would be necessary before any meaningful analysis could be performed.

---

## Submission Checklist

Before submitting, make sure you have:

- [ ] Completed all code cells in Parts 1 and 2
- [ ] Run all cells to show output
- [ ] Added your observations for each dataset
- [ ] Answered all reflection questions in Part 3
- [ ] Added your name and date at the top
- [ ] Saved the notebook
- [ ] Committed and pushed your changes to GitHub

**Great work completing your first R data analysis assignment! 🎉**

---

## 🚀 Ready to Submit?

### Easy Submission Steps (No Command Line Required!):

1. **Save this notebook** (Ctrl+S or File → Save)

2. **Use VS Code Source Control**:
   - Click the **Source Control** icon in the left sidebar (tree branch symbol)
   - Click the **"+"** button next to your notebook file
   - Type a message: `Submit homework 1 - [Your Name]`
   - Click **"Commit"** 
   - Click **"Sync Changes"** or **"Push"**

3. **Verify on GitHub**: Go to your repository online and confirm your notebook appears with your completed work

**📖 Need help?** See [GITHUB_CLASSROOM_SUBMISSION.md](../../GITHUB_CLASSROOM_SUBMISSION.md) for detailed instructions.

**🎉 Congratulations on completing your first R data analysis assignment!**